packages/babel-plugin-proposal-decorators/src/transformer-2023-05.ts

```diff
@@ -539,30 +556,65 @@ function transformClass(
     }
 
     if (element.node.decorators && element.node.decorators.length > 0) {
+      switch (element.node.type) {
+        case "ClassProperty":
+          // @ts-expect-error todo: propertyVisitor.ClassProperty should be callable. Improve typings.
+          propertyVisitor.ClassProperty(
+            element as NodePath<t.ClassProperty>,
+            state,
+          );
+          break;
+        case "ClassPrivateProperty":
+          // @ts-expect-error todo: propertyVisitor.ClassPrivateProperty should be callable. Improve typings.
+          propertyVisitor.ClassPrivateProperty(
+            element as NodePath<t.ClassPrivateProperty>,
+            state,
+          );
+          break;
+        case "ClassAccessorProperty":
+          // @ts-expect-error todo: propertyVisitor.ClassAccessorProperty should be callable. Improve typings.
+          propertyVisitor.ClassAccessorProperty(
+            element as NodePath<t.ClassAccessorProperty>,
+            state,
+          );
+          break;
+      }
       hasElementDecorators = true;
     } else if (element.node.type === "ClassAccessorProperty") {
+      // @ts-expect-error todo: propertyVisitor.ClassAccessorProperty should be callable. Improve typings.
+      propertyVisitor.ClassAccessorProperty(
+        element as NodePath<t.ClassAccessorProperty>,
+        state,
+      );
       const { key, value, static: isStatic, computed } = element.node;
 
       const newId = generateClassPrivateUid();
-
-      const valueNode = value ? t.cloneNode(value) : undefined;
-
-      const newField = generateClassProperty(newId, valueNode, isStatic);
+      const newField = generateClassProperty(newId, value, isStatic);
 
       const [newPath] = element.replaceWith(newField);
       addProxyAccessorsFor(
         path.node.id,
         newPath,
-        key,
+        computed && !scopeParent.isStatic(key)
+          ? memoiseExpression(key as t.Expression, "computedKey")
+          : key,
         newId,
         version,
         computed,
       );
     }
   }
 
-  // If nothing is decorated, return
-  if (!classDecorators && !hasElementDecorators) return;
+  // If nothing is decorated and no assignments inserted, return
+  if (!classDecorators && !hasElementDecorators) {
+    if (assignments.length > 0) {
```

Comment:
In this case, we don't have any decorators, but we should still transform the class accessors, the assignments at this time must be the memoized accessor keys.

Reply:
Can you add this as an in-code comment? :)